### ISSUE TITLE
Error Building my library in Windows

### ISSUE THREAD

**Issue by jftsaraiva** [jvandemo/generator-angular2-library#64]
Hi,

In Windows with a new library (only generated code) I'm getting (with `npm run build`):

```
0 info it worked if it ends with ok
1 verbose cli [ 'C:\\Program Files\\nodejs\\node.exe',
1 verbose cli   'C:\\Users\\myuser\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'run',
1 verbose cli   'build' ]
2 info using npm@4.5.0
3 info using node@v7.9.0
4 verbose run-script [ 'prebuild', 'build', 'postbuild' ]
5 info lifecycle angular-library@0.1.0~prebuild: angular-library@0.1.0
6 silly lifecycle angular-library@0.1.0~prebuild: no script for prebuild, continuing
7 info lifecycle angular-library@0.1.0~build: angular-library@0.1.0
8 verbose lifecycle angular-library@0.1.0~build: unsafe-perm in lifecycle true
9 verbose lifecycle angular-library@0.1.0~build: PATH: C:\Users\myuser\AppData\Roaming\npm\node_modules\npm\bin\node-gyp-bin;D:\work\WorkspaceLabs\angular-library\node_modules\.bin;C:\ProgramData\Oracle\Java\javapath;C:\Program Files\Microsoft MPI\Bin\;C:\Program Files (x86)\Intel\iCLS Client\;C:\Program Files\Intel\iCLS Client\;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\IPT;C:\Program Files\Intel\Intel(R) Management Engine Components\IPT;C:\Program Files (x86)\Skype\Phone\;C:\Program Files (x86)\Yarn\bin;C:\Users\myuser\.dnx\bin;C:\Program Files\Microsoft DNX\Dnvm\;C:\Program Files\Microsoft SQL Server\120\Tools\Binn\;C:\Program Files\Microsoft SQL Server\130\Tools\Binn\;C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit\;C:\Program Files\PuTTY\;C:\Program Files\dotnet\;C:\ProgramData\chocolatey\bin;C:\Program Files\OpenVPN\bin;C:\Program Files\Git\cmd;C:\Program Files (x86)\Microsoft SQL Server\130\Tools\Binn\;C:\Program Files (x86)\Microsoft SQL Server\130\DTS\Binn\;C:\Program Files\Microsoft SQL Server\130\DTS\Binn\;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\130\Tools\Binn\;C:\Program Files (x86)\Microsoft SQL Server\Client SDK\ODBC\130\Tools\Binn\;C:\Program Files (x86)\Microsoft SQL Server\130\Tools\Binn\ManagementStudio\;C:\Program Files\Intel\WiFi\bin\;C:\Program Files\Common Files\Intel\WirelessCommon\;C:\Program Files\nodejs\;C:\Users\myuser\AppData\Local\Microsoft\WindowsApps;C:\Program Files\Docker Toolbox;C:\Program Files (x86)\Microsoft VS Code\bin;C:\Users\myuser\AppData\Local\Yarn\.bin;C:\Users\myuser\AppData\Roaming\npm
10 verbose lifecycle angular-library@0.1.0~build: CWD: D:\work\WorkspaceLabs\angular-library
11 silly lifecycle angular-library@0.1.0~build: Args: [ '/d /s /c', './build.sh' ]
12 silly lifecycle angular-library@0.1.0~build: Returned: code: 1  signal: null
13 info lifecycle angular-library@0.1.0~build: Failed to exec build script
14 verbose stack Error: angular-library@0.1.0 build: `./build.sh`
14 verbose stack Exit status 1
14 verbose stack     at EventEmitter.<anonymous> (C:\Users\myuser\AppData\Roaming\npm\node_modules\npm\lib\utils\lifecycle.js:279:16)
14 verbose stack     at emitTwo (events.js:106:13)
14 verbose stack     at EventEmitter.emit (events.js:194:7)
14 verbose stack     at ChildProcess.<anonymous> (C:\Users\myuser\AppData\Roaming\npm\node_modules\npm\lib\utils\spawn.js:40:14)
14 verbose stack     at emitTwo (events.js:106:13)
14 verbose stack     at ChildProcess.emit (events.js:194:7)
14 verbose stack     at maybeClose (internal/child_process.js:899:16)
14 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
15 verbose pkgid angular-library@0.1.0
16 verbose cwd D:\work\WorkspaceLabs\angular-library
17 verbose Windows_NT 10.0.14393
18 verbose argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\myuser\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "run" "build"
19 verbose node v7.9.0
20 verbose npm  v4.5.0
21 error code ELIFECYCLE
22 error errno 1
23 error angular-library@0.1.0 build: `./build.sh`
23 error Exit status 1
24 error Failed at the angular-library@0.1.0 build script './build.sh'.
24 error Make sure you have the latest version of node.js and npm installed.
24 error If you do, this is most likely a problem with the angular-library package,
24 error not with npm itself.
24 error Tell the author that this fails on your system:
24 error     ./build.sh
24 error You can get information on how to open an issue for this project with:
24 error     npm bugs angular-library
24 error Or if that isn't available, you can get their info via:
24 error     npm owner ls angular-library
24 error There is likely additional logging output above.
25 verbose exit [ 1, true ]
````

I tried several versions of nodeJS and npm but the error always happens. 

The same command with the same src in a Linux environment: `Bash on Ubuntu on Windows` works great.

Any idea/clue about what may be causing the error in Windows?

**Comment by jdjuan** [comment]
So, the `bash.sh` file contains functions like `cp` and `rsync` which are not natively available in Windows. You can try installing [CyWin](https://cygwin.com) but I can't assure it will work. 

You can read #62 in which Gulp is considered as an alternative to avoid bash commands for Windows users.

**Comment by caroso1222** [comment]
You can even try this experimental gulpfile and give us some feedback https://github.com/jdjuan/ng-notyf/blob/master/gulpfile.js

**Comment by jvandemo** [comment]
I have added an experimental branch with Gulp that should support Mac, Linux and Windows (see #65).

@jftsaraiva Can you please install the Gulp version:

```bash
$ npm install -g https://github.com/jvandemo/generator-angular2-library#feat/gulp
```

and verify if everything works as expected?

After generating a library, you should be able to build your library using:

```
$ npm run build
```

or use watch mode to build whenever a file changes in `src`:


```
$ npm run build:watch
```

Can you please verify if everything works as expected before we merge changes into master?

**Comment by jftsaraiva** [comment]
In Windows worked fine.

**Comment by jvandemo** [comment]
@jftsaraiva &mdash; Awesome, thank you for the confirmation. I have released the gulp version as v9.0.0. 

Feel free to let us know if you experience any issues.
